### PR TITLE
Build all tests and examples from the main CMakeLists.txt and use ctest for test execution

### PR DESCRIPTION
**Pull request by BenjaminW3**
This makes sure that #537 does not break again and that I can start with #543.

### CODE REVIEW

File: script/travis/run_tests.sh [hunk 1]
```diff
@@ -30,42 +30,24 @@ set -vuo pipefail
...
+
+    #-------------------------------------------------------------------------------
+    # Execute all examples.
+    # NOTE: The examples are hard-coded to use a CPU accelerator which is not available in ALPAKA_ACC_GPU_CUDA_ONLY_MODE
+    if [ "${ALPAKA_ACC_GPU_CUDA_ONLY_MODE}" == "OFF" ]
+    then
+        cd "example/bufferCopy" && ./bufferCopy && cd ../..
+        cd "example/helloWorld" && ./helloWorld && cd ../..
+        cd "example/helloWorldLambda" && ./helloWorldLambda && cd ../..
+        cd "example/vectorAdd" && ./vectorAdd && cd ../..
```
Comment (BenjaminW3) [inline script/travis/run_tests.sh]: I will add the examples to the tests in a separate PR.

File: include/alpaka/exec/ExecGpuCudaRt.hpp [hunk 2]
```diff
@@ -496,7 +496,7 @@ namespace alpaka
                             exec::cuda::detail::cudaKernel<TDim, TIdx, TKernelFnObj, TArgs...><<<
                                 gridDim,
                                 blockDim,
-                                blockSharedMemDynSizeBytes,
```
Comment (psychocoderHPC) [inline include/alpaka/exec/ExecGpuCudaRt.hpp]: Is this a bugfix? If so we should move it to a separate PR because this change not fits to the description.

Comment (BenjaminW3) [reply include/alpaka/exec/ExecGpuCudaRt.hpp]: This fixes a warning that only occurs when ALPAKA_DEBUG==2 and was an error with Werror. I could create a new PR with the fix if you want.

Comment (psychocoderHPC) [reply include/alpaka/exec/ExecGpuCudaRt.hpp]: That would be nice because this is much easier to back-port.

Comment (BenjaminW3) [reply include/alpaka/exec/ExecGpuCudaRt.hpp]: The fix has been merged separately

**Comment by BenjaminW3** [comment]
Ready to be merged!

**Comment by BenjaminW3** [comment]
Anything missing for this to be approved?
